controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -464,18 +458,36 @@ func (r *StorageConsumerReconciler) reconcileCephRadosNamespace(
 		}
 		rns.Namespace = r.namespace
 
-		if _, err := ctrl.CreateOrUpdate(r.ctx, r.Client, rns, func() error {
-			if err := controllerutil.SetControllerReference(r.storageConsumer, rns, r.Scheme); err != nil {
-				return err
+		if bp.DeletionTimestamp.IsZero() {
+			if _, err := ctrl.CreateOrUpdate(r.ctx, r.Client, rns, func() error {
+				if err := controllerutil.SetControllerReference(r.storageConsumer, rns, r.Scheme); err != nil {
+					return err
+				}
+				if err := controllerutil.SetOwnerReference(additionalOwner, rns, r.Scheme); err != nil {
+					return err
+				}
+				rns.Spec.Name = radosNamespaceName
+				rns.Spec.BlockPoolName = bp.Name
+				return nil
+			}); err != nil {
+				multierr.AppendInto(&combinedErr, err)
 			}
-			if err := controllerutil.SetOwnerReference(additionalOwner, rns, r.Scheme); err != nil {
-				return err
+		} else {
+			if bp.Status != nil {
+				for idx := range bp.Status.Conditions {
+					condition := &bp.Status.Conditions[idx]
+					// block pool is not waiting for rbd images
+					if condition.Type == rookCephv1.ConditionPoolDeletionIsBlocked &&
+						condition.Status == corev1.ConditionFalse {
+						if err := r.Delete(r.ctx, rns); client.IgnoreNotFound(err) != nil {
+							multierr.AppendInto(&combinedErr, err)
+						}
+						break
```

Comment:
The loop is a search loop and should only concern itself with finding the item in the list. Handling of the item should be handled after the loop

Reply:
done.